flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/resources/TestingLeaderCallbackHandler.java

```diff
@@ -63,38 +59,23 @@ public boolean hasLeadership() {
         return isLeader;
     }
 
-    public static String waitUntilNewLeaderAppears(long timeout) throws Exception {
-        final AtomicReference<String> leaderRef = new AtomicReference<>();
-        CommonTestUtils.waitUntilCondition(
-                () -> {
-                    final String lockIdentity = sharedQueue.poll(timeout, TimeUnit.MILLISECONDS);
-                    leaderRef.set(lockIdentity);
-                    return lockIdentity != null;
```

Comment:
I was wondering whether we should add a null check in the constructor to comply to the previous implementation (ignoring `null` values). But `BlockingQueue`s don't accept `null` as values, anyway. The only benefit we gain from it is that we would fail earlier (in the constructor rather than the `offer` calls) if somebody "accidentally" used `null` as a `lockIdentity`. but that's probably off-topic to this change, anyway. So, I leave it up to you...